include/zephyr/drivers/video-controls.h

```diff
@@ -31,42 +38,91 @@ extern "C" {
 
 /**
  * @name Control classes
+ *
+ * List of control classes, to group related controls together in blocks.
+ *
+ * This list is kept identical to the Linux kernel definitions.
+ *
  * @{
  */
-#define VIDEO_CTRL_CLASS_GENERIC	0x00000000	/**< Generic class controls */
-#define VIDEO_CTRL_CLASS_CAMERA		0x00010000	/**< Camera class controls */
-#define VIDEO_CTRL_CLASS_MPEG		0x00020000	/**< MPEG-compression controls */
-#define VIDEO_CTRL_CLASS_JPEG		0x00030000	/**< JPEG-compression controls */
-#define VIDEO_CTRL_CLASS_VENDOR		0xFFFF0000	/**< Vendor-specific class controls */
+#define VIDEO_CTRL_CLASS_BASE			0x00980900 /**< Base class controls */
+#define VIDEO_CTRL_CLASS_CODEC			0x00990900 /**< Stateful codec controls */
+#define VIDEO_CTRL_CLASS_CAMERA			0x009a0900 /**< Camera class controls */
+#define VIDEO_CTRL_CLASS_FM_TX			0x009b0900 /**< FM Modulator controls */
+#define VIDEO_CTRL_CLASS_FLASH			0x009c0900 /**< Camera flash controls */
+#define VIDEO_CTRL_CLASS_JPEG			0x009d0900 /**< JPEG-compression controls */
+#define VIDEO_CTRL_CLASS_IMAGE_SOURCE		0x009e0900 /**< Image source controls */
+#define VIDEO_CTRL_CLASS_IMAGE_PROC		0x009f0900 /**< Image processing controls */
+#define VIDEO_CTRL_CLASS_DV			0x00a00900 /**< Digital Video controls */
```

Comment:
From line 56 to line 62, I think we should not add class that are actually not used by any sensors, especially these classes will be rarely used. When needed, it can be added later

Reply:
I made an exception for the classes: provide them all as in Linux, but thinking again, I see no reason for that exception.
It makes more sense as you suggested, will remove. Thank you!

Reply:
Removed from 56 to 61, but keeping the vendor control class that can be used by an application.
Did you mean to remove it also? I would try use it for USB UVC which also has a vendor control section.
I can of course wait to introduce something that uses it before adding the class!

Reply:
Yes, it's ok to keep the vendor class. I just meant the classes that are not currently used by any sensor and will be rarely used in the future.